environs/errors.go

```diff
@@ -41,3 +43,19 @@ func IsAvailabilityZoneIndependent(err error) bool {
 	}
 	return false
 }
+
+type PreferredStorageNotFound struct {
```

Comment:
comment these
plus they belong with the k8s stuff

Reply:
I don't think this is correct. It's an error that makes sense for many providers.

Reply:
Only used for k8s now but it could be used elsewhere in future, I have no strong objections to it being here.